packages/aws-cdk-lib/aws-apigatewayv2/lib/websocket/integration-response.ts

```diff
@@ -0,0 +1,194 @@
+import { IResource, Resource } from '../../../core';
+import { CfnIntegrationResponse } from '../apigatewayv2.generated';
+
+/**
+ * WebSocket integration response key helper class
+ */
+export class WebSocketIntegrationResponseKey {
+  /**
+   * Match all responses
+   */
+  public static default = new WebSocketIntegrationResponseKey('$default');
+
+  /**
+   * Match all 2xx responses (HTTP success codes)
+   */
+  public static success = WebSocketIntegrationResponseKey.fromStatusRegExp(/2\d{2}/.source);
+
+  /**
+   * Match all 4xx responses (HTTP client error codes)
+   */
+  public static clientError = WebSocketIntegrationResponseKey.fromStatusRegExp(/4\d{2}/.source);
+
+  /**
+   * Match all 5xx responses (HTTP server error codes)
+   */
+  public static serverError = WebSocketIntegrationResponseKey.fromStatusRegExp(/5\d{2}/.source);
+
+  /**
+   * Generate an integration response key from an HTTP status code
+   *
+   * @example
+   * // Match 403 status code
+   * apigwv2.WebSocketIntegrationResponseKey.fromStatusCode(403)
+   *
+   * @param httpStatusCode HTTP status code of the mapped response
+   */
+  public static fromStatusCode(httpStatusCode: number): WebSocketIntegrationResponseKey {
+    return new WebSocketIntegrationResponseKey(`/${httpStatusCode}/`);
+
+  }
+
+  /**
+   * Generate an integration response key from a regular expression matching HTTP status codes
+   *
+   * @example
+   * // Match all HTTP client and server error status codes
+   * apigwv2.WebSocketIntegrationResponseKey.fromStatusRegExp('4\\d{2}|5\\d{2}')
+   *
+   * // Match all HTTP client and server error status codes,
+   * // using the RegExp built-in object
+   * apigwv2.WebSocketIntegrationResponseKey.fromStatusRegExp(/4\d{2}|5\d{2}/.source)
+   *
+   * @param httpStatusRegExpStr HTTP status code regular expression string representation
+   * @see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
+   *
+   * @throws an error if {@link httpStatusRegExpStr} is not a valid regular expression string
+   */
+  public static fromStatusRegExp(httpStatusRegExpStr: string): WebSocketIntegrationResponseKey {
+    const httpStatusRegExp = new RegExp(httpStatusRegExpStr);
+
+    return new WebSocketIntegrationResponseKey(`/${httpStatusRegExp.source}/`);
+  }
+
+  /**
+   * WebSocket integration response private constructor
+   *
+   * @param key The key of the integration response
+   */
+  private constructor(readonly key: string) {}
+
+  /** String representation of the integration response key */
+  public toString(): string {
+    return this.key;
+  }
+}
+
+/**
+ * WebSocket integration response properties, used internally for Integration implementations
+ * The integration will add itself these props during the bind process
+ *
+ * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-integration-responses.html
+ */
+export interface InternalWebSocketIntegrationResponseProps {
```

Comment:
Instead of making this a props interface, can you make this a class with the first input in the constructor being the `responseKey` and then the rest props? I think that will be a bit cleaner than the current experience.